lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -32,53 +36,115 @@ def route(verb, action, *)
         end
 
         def self.registered(app)
-          ::Sinatra::Base.module_eval do
-            def render(engine, data, *)
-              output = ''
-              tracer = Datadog.configuration[:sinatra][:tracer]
-              if tracer.enabled
-                tracer.trace(Ext::SPAN_RENDER_TEMPLATE, span_type: Datadog::Ext::HTTP::TEMPLATE) do |span|
-                  # If data is a string, it is a literal template and we don't
-                  # want to record it.
-                  span.set_tag(Ext::TAG_TEMPLATE_NAME, data) if data.is_a? Symbol
-
-                  # Measure service stats
-                  Contrib::Analytics.set_measured(span)
-
-                  output = super
-                end
-              else
-                output = super
-              end
+          app.use TracerMiddleware
 
-              output
-            end
-          end
+          app.after do
+            configuration = Datadog.configuration[:sinatra]
+            return unless configuration[:tracer].enabled
+            return if Sinatra::Env.middleware_traced?(env)
```

Comment:
This prevents all the middleware from being shown, correct? 

Also, is `return` safe to do here? Sometimes blocks don't do `return` well.

Reply:
Good catching, changing these to `next`.